homeassistant/components/mobile_app/const.py

```diff
@@ -91,6 +91,7 @@
     vol.Required(ATTR_APP_ID): cv.string,
     vol.Required(ATTR_APP_NAME): cv.string,
     vol.Required(ATTR_APP_VERSION): cv.string,
+    vol.Required(ATTR_DEVICE_ID): cv.string,
```

Comment:
Maybe we should extend this schema from the registration update schema since there's an overlap?

Reply:
Done

Reply:
Why won't we just derive this automatically from name  ?

Reply:
And do that only once, during registration, after that use that.

Reply:
This is all a stopgap anyway, as once the device tracker has been reworked, users can rename entity IDs.

Reply:
This should not be allowed to be part of the update schema, because it doesn't work like that.

All mobile app entities have a unique ID, so that means that when the system sees them first, `self.entity_id` will be taken as a suggestion, adjusted if it is already taken, and then that's it. After that, `self.entity_id` is overridden by whatever the entity registry tells us the entity is